warehouse/ingest-core/src/main/java/datawave/iterators/FrequencyColumnIterator.java

```diff
@@ -0,0 +1,186 @@
+package datawave.iterators;
+
+import datawave.data.ColumnFamilyConstants;
+import datawave.query.util.Frequency;
+import datawave.query.util.FrequencyFamilyCounter;
+import datawave.query.util.MetadataHelper;
+import datawave.query.util.YearMonthDay;
+import org.apache.accumulo.core.conf.AccumuloConfiguration;
+import org.apache.accumulo.core.data.Key;
+import org.apache.accumulo.core.data.PartialKey;
+import org.apache.accumulo.core.data.Value;
+import org.apache.accumulo.core.iterators.IteratorEnvironment;
+import org.apache.accumulo.core.iterators.IteratorUtil;
+import org.apache.accumulo.core.iterators.SortedKeyValueIterator;
+import org.apache.accumulo.core.iterators.user.TransformingIterator;
+import org.apache.accumulo.core.security.Authorizations;
+import org.apache.accumulo.core.security.VisibilityEvaluator;
+import org.apache.commons.collections.map.LRUMap;
+import org.apache.hadoop.io.Text;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class FrequencyColumnIterator extends TransformingIterator {
+    
+    private FrequencyFamilyCounter frequencyFamilyCounter;
+    // TODO Figure out how to keep the rowIdToCompressedFreqCQMap from getting too large
+    // TODO maybe we do not have to worry about that from happening.
+    private HashMap<String,FrequencyFamilyCounter> rowIdToCompressedFreqCQMap = new HashMap<>();
+    private String ageOffDate = "20100101";
```

Comment:
if you are going to have a default, it better be pretty old e.g. "19000101"

Reply:
Done.